platform/core/src/services/UIContextMenuService/index.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * UI Context Menu
+ *
+ * @typedef {Object} ContextMenuProps
+ * @property {Event} event The event with tool information.
+ */
+
+const uiContextMenuServicePublicAPI = {
+  name: 'UIContextMenuService',
+  hide,
+  show,
+  setServiceImplementation,
+};
+
+const uiContextMenuServiceImplementation = {
+  _show: () => console.warn('show() NOT IMPLEMENTED'),
+  _hide: () => console.warn('hide() NOT IMPLEMENTED'),
+};
+
+function createUIContextMenuService() {
+  return uiContextMenuServicePublicAPI;
+}
+
+/**
+ * Show a new UI ContextMenu dialog;
+ *
+ * @param {ContextMenuProps} props { event }
+ */
+function show({ event }) {
+  return uiContextMenuServiceImplementation._show({
+    event,
+  });
+}
+
+/**
+ * Hide a UI ContextMenu dialog;
+ *
+ */
+function hide() {
+  return uiContextMenuServiceImplementation._hide();
+}
+
+/**
+ *
+ *
+ * @param {*} {
+ *   show: showImplementation,
+ *   hide: hideImplementation,
+ * }
+ */
+function setServiceImplementation({
+  show: showImplementation,
+  hide: hideImplementation,
+}) {
+  if (showImplementation) {
+    uiContextMenuServiceImplementation._show = showImplementation;
+  }
+  if (hideImplementation) {
+    uiContextMenuServiceImplementation._hide = hideImplementation;
+  }
+}
+
+export default createUIContextMenuService;
```

Comment:
`UIContextMenuService` and `UILabelingFlowService` look similar to Dialog. I'm wondering if there's a smart way to say these a flavored version of `UIDialogService`?

I'll keep reading, but we should at least think about other ways to structure this.

Reply:
Yeah, they have the same service/factory interface... we can have the dialog service generic enough to be used to instantiate dialogs and flavored versions of dialogs but this wouldnt be flexible. Maybe migrate to class and use extends, something like BaseTool in cornerstone?